src/test/java/io/confluent/examples/streams/StreamToStreamJoinIntegrationTest.java

```diff
@@ -112,10 +111,19 @@ public void shouldJoinTwoStreams() throws Exception {
     // for the same join key (e.g. "newspaper-advertisement"), we receive an update from either of
     // the two joined streams during the defined join window.
     final KStream<String, String> impressionsAndClicks = alerts.outerJoin(
-        incidents,
-        (impressionValue, clickValue) -> impressionValue + "/" + clickValue,
-        // KStream-KStream joins are always windowed joins, hence we must provide a join window.
-        JoinWindows.of(TimeUnit.SECONDS.toMillis(5))
+      incidents,
+      (impressionValue, clickValue) ->
+        (clickValue == null)? impressionValue + "/not-clicked-yet": impressionValue + "/" + clickValue,
+      // KStream-KStream joins are always windowed joins, hence we must provide a join window.
+      JoinWindows.of(Duration.ofSeconds(5)),
+      // In this specific example, we don't need to define join serdes explicitly because the key, left value, and
+      // right value are all of type String, which matches our default serdes configured for the application.  However,
+      // we want to showcase the use of `Joined.with(...)` in case your code needs a different type setup.
+      Joined.with(
```

Comment:
Don't think we need this -- default serdes in configs is `StringSerde` anyway.

Reply:
I'd like to keep this included because we often get questions from people because they joins failed due to serdes not having being specified, and it's not obvious when/where this is needed.